pkg/controller/oneagent/istio.go

```diff
@@ -84,37 +102,79 @@ func (r *ReconcileOneAgent) reconcileIstioRemoveConfigurations(instance *dynatra
 		seen[istio.BuildNameForEndpoint(instance.Name, ch.Host, ch.Port)] = true
 	}
 
-	vsUpd := r.reconcileIstioRemoveConfiguration(instance, istio.VirtualServiceGVK, listOps, seen, logger)
-	seUpd := r.reconcileIstioRemoveConfiguration(instance, istio.ServiceEntryGVK, listOps, seen, logger)
+	vsUpd := r.reconcileIstioRemoveConfiguration(ic, istio.VirtualServiceGVK, listOps, seen, logger)
+	seUpd := r.reconcileIstioRemoveConfiguration(ic, istio.ServiceEntryGVK, listOps, seen, logger)
 
 	return vsUpd || seUpd
 }
 
-func (r *ReconcileOneAgent) reconcileIstioRemoveConfiguration(instance *dynatracev1alpha1.OneAgent, gvk schema.GroupVersionKind,
-	listOps *client.ListOptions, seen map[string]bool, logger logr.Logger) bool {
+func (r *ReconcileOneAgent) initialiseIstioClient(logger logr.Logger) (*versionedIstioClient.Clientset, error) {
+	ic, err := versionedIstioClient.NewForConfig(r.config)
+	if err != nil {
+		logger.Error(err, fmt.Sprint("istio: failed to initialise client"))
+	}
+	return ic, err
+}
 
-	var list unstructured.UnstructuredList
-	list.SetGroupVersionKind(gvk)
+func (r *ReconcileOneAgent) reconcileIstioRemoveConfiguration(
+	ic *versionedIstioClient.Clientset,
+	gvk schema.GroupVersionKind,
+	listOps *client.ListOptions,
```

Comment:
`listOps` is not used. IIRC, it had some labels to differentiate communication endpoints from different sets. Does this logic work?